PROJ is not able to be built as a subdirectory

<!-- Please search existing issues to avoid creating duplicates. -->


#### Example of problem

```
#include <proj.h>
#include <iostream>
int main(){
    return 0;
}
```

```
add_subdirectory(external/PROJ)
add_executable(test main.cpp)
target_link_libraries(test PRIVATE PROJ::proj)
```
```
-- PROJ: Configured 'dist' target
-- Configuring done
  Target "test" links to target "PROJ::proj" but the target was not found.
  Perhaps a find_package() call is missing for an IMPORTED target, or an
  ALIAS target is missing?
```

with out alias target


```
add_subdirectory(external/PROJ)
add_executable(test main.cpp)
target_link_libraries(test PRIVATE proj) 
```

/home/user/projects/test/main.cpp:2:10: fatal error: proj.h: No such file or directory
 #include <proj.h>
          ^~~~~~~~

#### Problem description
Using PROJ as a sub directory is not possible, this creates issues with many workflows, especially if one is not using a package manager with PROJ, which is very likely now that VCPKG only supports pre 8.0 versions of proj and does not support geotiff builds. 

PROJ currently creates an alias target for `PROJ::proj`, but only during installs, and it doesn't appear as if any usable target, however this is not a big deal as long as *a* target exists.  Currently it appears as if `proj` is that target, however it doesn't appear to properly propagate it's own `target_include_directories`, so it can't be used in isolation. 

#### Expected Output

The above example should compile with out error. 

#### Environment Information
 - PROJ version 8.2.0
 - Operation System Information
 Centos 7
#### Installation method
 - from sourfce

--- a/src/lib_proj.cmake
+++ b/src/lib_proj.cmake
@@ -372,6 +372,8 @@ add_library(proj
   ${ALL_LIBPROJ_HEADERS}
   ${PROJ_RESOURCES}
 )
+add_library(PROJ::proj ALIAS proj)
+
 target_compile_options(proj
   PRIVATE $<$<COMPILE_LANGUAGE:C>:${PROJ_C_WARN_FLAGS}>
   PRIVATE $<$<COMPILE_LANGUAGE:CXX>:${PROJ_CXX_WARN_FLAGS}>
@@ -402,6 +404,8 @@ if(ENABLE_IPO)
 endif()
 
 target_include_directories(proj INTERFACE
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
+  $<BUILD_INTERFACE:${PROJ_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:${INCLUDEDIR}>)
 
 if(WIN32)

--- a/travis/install.sh
+++ b/travis/install.sh
@@ -115,10 +115,33 @@ fi
 
 if [ "$BUILD_NAME" != "linux_gcc8" -a "$BUILD_NAME" != "linux_gcc_32bit" ]; then
 
-    # cmake build from generated tarball
+    cmake --version
+
+    # Build PROJ as a subproject
+    mkdir proj_as_subproject
+    cd proj_as_subproject
+    mkdir external
+    ln -s $PWD/.. external/PROJ
+
+    echo '#include "proj.h"' > mytest.c
+    echo 'int main() { proj_info(); return 0; }' >> mytest.c
+
+    echo 'cmake_minimum_required(VERSION 3.9)' > CMakeLists.txt
+    echo 'project(mytest)' >> CMakeLists.txt
+    echo 'add_subdirectory(external/PROJ)' >> CMakeLists.txt
+    echo 'add_executable(mytest mytest.c)' >> CMakeLists.txt
+    echo 'target_include_directories(mytest PRIVATE $<TARGET_PROPERTY:PROJ::proj,INTERFACE_INCLUDE_DIRECTORIES>)' >> CMakeLists.txt
+    echo 'target_link_libraries(mytest PRIVATE PROJ::proj)' >> CMakeLists.txt
+
     mkdir build_cmake
     cd build_cmake
-    cmake --version
+    cmake .. -DCMAKE_BUILD_TYPE=Debug
+    VERBOSE=1 make >/dev/null
+    cd ../..
+
+    # Regular build
+    mkdir build_cmake
+    cd build_cmake
     cmake .. -DCMAKE_INSTALL_PREFIX=/tmp/proj_cmake_install
     VERBOSE=1 make >/dev/null
     make install >/dev/null

